src/Api/ExceptionHandler/FallbackExceptionHandler.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * (c) Toby Zerner <toby.zerner@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Flarum\Api\ExceptionHandler;
+
+use Exception;
+use Psr\Log\LoggerInterface;
+use Tobscure\JsonApi\Exception\Handler\ExceptionHandlerInterface;
+use Tobscure\JsonApi\Exception\Handler\ResponseBag;
+
+class FallbackExceptionHandler implements ExceptionHandlerInterface
+{
+    /**
+     * @var bool
+     */
+    protected $debug;
+
+    /**
+     * @var LoggerInterface
+     */
+    protected $logger;
+
+    /**
+     * @param bool $debug
+     * @param LoggerInterface $logger
+     */
+    public function __construct($debug, LoggerInterface $logger)
+    {
+        $this->debug = $debug;
+        $this->logger = $logger;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function manages(Exception $e)
+    {
+        return true;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(Exception $e)
+    {
+        $status = 500;
```

Comment:
Some Illuminate Exceptions allow setting the HTTP status code as well. These can be found as implementations of the HttpExceptions from Symfony. Forcing uncaught exceptions to be shown as 500 might be conflicting with the reason of the author, eg Unauthorized Exceptions should become 401, whereas Forbidden Exceptions should become 403. We should at least try to see if the Exception has a method `getStatusCode()` and use that.

Reply:
This is what the `Api\ExceptionHandler`s are for. If an exception is thrown that we haven't accounted for in an ExceptionHandler, then it should count as a 500 Internal Server Error regardless, because we don't necessarily want to expose what kind of exception/status code it is. Plus, just using the status code would be awkward because the message would still be "internal server error".